src/CMakeLists.txt

```diff
@@ -44,7 +43,9 @@ elseif(UMF_BUILD_LEVEL_ZERO_PROVIDER)
     message(STATUS "Level Zero include directory: ${LEVEL_ZERO_INCLUDE_DIRS}")
 endif()
 
-if(UMF_BUILD_CUDA_PROVIDER)
+# Fetch CUDA only if needed i.e.: if building CUDA provider is ON and CUDA
+# headers are not provided by the user (via setting UMF_CUDA_INCLUDE_DIR).
```

Comment:
`UMF_CUDA_INCLUDE_DIR` is not a `CACHE` var (it's not defined, as `UMF_LEVEL_ZERO_INCLUDE_DIR` on top)

Reply:
Done